src/detect-msg.c

```diff
@@ -125,89 +125,66 @@ static int DetectMsgSetup (DetectEngineCtx *de_ctx, Signature *s, const char *ms
 #ifdef UNITTESTS
 static int DetectMsgParseTest01(void)
 {
-    int result = 0;
     Signature *sig = NULL;
     const char *teststringparsed = "flow stateless to_server";
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL)
-        goto end;
+    FAIL_IF(de_ctx == NULL);
```

Comment:
We have a slightly cleaner macro for checking pointers: `FAIL_IF_NULL(ptr)`.

Reply:
Understood! I saw that option at first, but wasn't really sure it was ready for using, because of the issue open that seems to be related to that...